modules/programs/alacritty.nix

```diff
@@ -0,0 +1,47 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.alacritty;
+
+in
+
+{
+  options = {
+    programs.alacritty = {
+      enable = mkEnableOption "Alacritty";
+
+      settings = mkOption {
+        type = types.attrs;
+        default = {};
+        example = literalExample ''
+          {
+            window.dimensions = {
+              lines = 3;
+              columns = 200;
+            };
+            key_bindings = [
+              {
+                key = "K";
+                mods = "Control";
+                chars = "\\x0c";
+              }
+            ];
+          }
+        '';
+        description = ''
+          Configuration written to <filename>~/.config/alacritty/alacritty.yml</filename>.
+          See <link xlink:href="https://github.com/jwilm/alacritty/blob/master/alacritty.yml">the default config file</link> for possible values.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.alacritty ];
+
+    xdg.configFile."alacritty/alacritty.yml".text = replaceStrings ["\\\\"] ["\\"] (builtins.toJSON cfg.settings);
```

Comment:
Can break this long line. Something like
```
     xdg.configFile."alacritty/alacritty.yml".text =
       replaceStrings ["\\\\"] ["\\"] (builtins.toJSON cfg.settings);
```
for example.

Reply:
@hyperfekt do you happen to remember why the backslash replacement was necessary?

Reply:
@ncfavier https://github.com/NixOS/nixpkgs/issues/97310
TL:DR; Nix strings don't support hex/unicode escapes, so a group of `\\` is retained by replacing the escaped backslashes (from the `toJSON` step) to their original form.